stream_alert/rule_processor/parsers.py

```diff
@@ -398,6 +398,40 @@ class KVParser(ParserBase):
     __default_separator = '='
     __default_delimiter = ' '
 
+    @staticmethod
+    def _string_to_list(data, delimiter, separator):
```

Comment:
Hey @jacknagz I'm curious if you looked at using `shlex` for this type of splitting? I think we can avoid rolling all of this logic on our own. For instance, does this provide what you're looking for? With the test data you provide of `'name=test result=OK date="Friday, January 01, 2018" count=5'`:

```
>>> import shlex
>>> shlex.split('name=test result=OK date="Friday, January 01, 2018" count=5')
['name=test', 'result=OK', 'date=Friday, January 01, 2018', 'count=5']
```

Reply:
@ryandeivert  Good call, I didn't know about that library. I'll add it and re-commit.

Reply:
👍 as an aside, you may want to cast to `str` first or use `ushlex` instead to enable better unicode support

Reply:
Or you can convert everything to Python3 for better unicode support :trollface:

Reply:
trollolololol